### PR TITLE
fix: sync engine integration tests [WPB-6480]

### PR DESCRIPTION
**Pull request by netbe**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6480" title="WPB-6480" target="_blank">Sub-taskWPB-6480</a>  [iOS] update Integration Tests for SyncEngine
  </td></table>
  <br />
 

# What's new in this PR?

This PR fixes tests on sync engine integration tests target

### CODE REVIEW

File: wire-ios-sync-engine/Source/UserSession/ZMUserSession+RecurringAction.swift [hunk 1]
```diff
@@ -35,7 +35,7 @@ extension ZMUserSession {
 
     var refreshUsersMissingMetadataAction: RecurringAction {
         .init(id: #function, interval: 3 * .oneHour) { [weak self] in
-
+            // TODO: check why do we refreshData on main and block main thread here? [WPB-6737]
```
Comment (netbe) [inline wire-ios-sync-engine/Source/UserSession/ZMUserSession+RecurringAction.swift]:
```suggestion
            // TODO: [WPB-6737] check why do we refreshData on main and block main thread here?
```

File: wire-ios-sync-engine/WireSyncEngine.xcodeproj/xcshareddata/xcschemes/WireSyncEngine.xcscheme [hunk 2]
```diff
@@ -128,7 +128,7 @@
          </CommandLineArgument>
          <CommandLineArgument
             argument = "-com.apple.CoreData.ConcurrencyDebug 1"
-            isEnabled = "NO">
+            isEnabled = "YES">
```
Comment (markusfassbender) [inline wire-ios-sync-engine/WireSyncEngine.xcodeproj/xcshareddata/xcschemes/WireSyncEngine.xcscheme:131]: Should we enable this on default? Does it have impact for production?

**Review by markusfassbender — APPROVED** [review]

File: wire-ios-sync-engine/WireSyncEngine.xcodeproj/xcshareddata/xcschemes/WireSyncEngine.xcscheme (hunk 2 above)
Comment (netbe) [reply wire-ios-sync-engine/WireSyncEngine.xcodeproj/xcshareddata/xcschemes/WireSyncEngine.xcscheme:131]: No impact on prod just when you run the app

File: wire-ios-sync-engine/Tests/Source/Synchronization/ZMHotFixTests.swift [hunk 3]
```diff
@@ -246,44 +246,44 @@ final class ZMHotFixTests_Integration: MessagingTest {
     }
 
     func testThatItUpdatesAccessRolesForConversations_432_1_0() {
```
Comment (markusfassbender) [inline wire-ios-sync-engine/Tests/Source/Synchronization/ZMHotFixTests.swift:248]: This flaky test is (hopefully) fixed now: the flaky cause seemed to add the notification observer after performing already the logic. 
Fix: the observer is added first.

**Review by markusfassbender — APPROVED** [review]

**Review by agisilaos — APPROVED** [review]

**Pull request merged**